### PR TITLE
Add int testing to bytes_and_flops

### PR DESCRIPTION
**Pull request by crtrott**
Also split the compilation into 4 source files

### CODE REVIEW

**Comment by crtrott** [comment]
CUDA compile time on my machine went from 47s to 26s, even though I added int instantiation.

File: benchmarks/bytes_and_flops/bench_double.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// ************************************************************************
+//@HEADER
+*/
+
+#include <bench.hpp>
+
+void run_stride_unroll_double(int N, int K, int R, int D, int U, int F, int T,
+                              int S) {
+  run_stride_unroll<double>(N, K, R, D, U, F, T, S);
+}
```
Comment (dalg24) [inline benchmarks/bytes_and_flops/bench_double.cpp:50]:
```suggestion
template void run_stride_unroll<double>(int N, int K, int R, int D, int U, int F, int T, int S);
```
(not formatted and not double checked the signature)

**Review by dalg24 — CHANGES_REQUESTED** [review]
Do not define new functions, just explicitly instantiate them

File: benchmarks/bytes_and_flops/main.cpp [hunk 2]
```diff
@@ -44,15 +44,17 @@
 
 #include <Kokkos_Core.hpp>
 #include <Kokkos_Timer.hpp>
-#include <bench.hpp>
 #include <cstdlib>
 
+template <class T>
+void run_stride_unroll(int, int, int, int, int, int, int, int);
```
Comment (dalg24) [inline benchmarks/bytes_and_flops/main.cpp]: No.  Keep the header include and do `extern template ...` to skip the instantiation from that compile unit.
(I pushed a fix directly to your branch)

**Review by dalg24 — APPROVED** [review]

**Comment by crtrott** [comment]
Retest this please.

**Pull request merged**